test/testdrive/replica-expiration.td

```diff
@@ -141,13 +206,67 @@ ALTER SYSTEM SET compute_replica_expiration_offset = 0;
     content TEXT,
     event_ts TIMESTAMP
   );
-> CREATE VIEW last_30_days AS
+> CREATE VIEW events_view AS
   SELECT event_ts, content
   FROM events
   WHERE mz_now() <= event_ts + INTERVAL '30 days';
-> CREATE DEFAULT INDEX ON last_30_days;
+> CREATE DEFAULT INDEX ON events_view;
 > INSERT INTO events SELECT x::text, now() FROM generate_series(1, 1000) AS x;
-> SELECT records FROM mz_introspection.mz_arrangement_sizes mas JOIN mz_introspection.mz_dataflow_operators mdo ON mas.operator_id = mdo.id AND name = 'ArrangeBy[[Column(0), Column(1)]]';
+> SELECT records FROM mz_introspection.mz_dataflow_arrangement_sizes
+  WHERE name LIKE '%events_view_primary_idx';
 2000
 > DROP TABLE events CASCADE;
 > DROP CLUSTER test;
+
+
+# Test that a constant collection is not expired on a cluster with replication factor 0
+
+$ postgres-execute connection=postgres://mz_system:materialize@${testdrive.materialize-internal-sql-addr}
+ALTER SYSTEM SET compute_replica_expiration_offset = '30d';
+> CREATE CLUSTER test (SIZE = '1', REPLICATION FACTOR = 0);
+> SET CLUSTER TO test;
+# `now()` cannot be materialized.
+> CREATE MATERIALIZED VIEW events_mv AS
+  SELECT x::text AS content, '2024-10-09 07:05:10.318+00'::timestamptz AS event_ts
+  FROM generate_series(1, 1000) AS x;
+> CREATE VIEW events_view AS
+  SELECT event_ts, content
+  FROM events_mv
+  WHERE mz_now() <= event_ts + INTERVAL '30 years';
+> CREATE DEFAULT INDEX ON events_view;
+> ALTER CLUSTER test SET (REPLICATION FACTOR = 1);
+> SELECT count(*) FROM events_view;
+1000
+> DROP MATERIALIZED VIEW events_mv CASCADE;
+> DROP CLUSTER test CASCADE;
+
+
+# Test that retraction diffs within the expiration time are generated after the replica expires and panics
```

Comment:
I fear this test will be flaky as-is. The timing requirements are tight, and that usually doesn't work well in CI. Could we find a way to make it less time-dependent?

Reply:
We could increase the time ranges and sleep accordingly.

Reply:
Increased expiration time to 30s.